### PR TITLE
Fix Binder and Colab links on Github (#1469)

### LINKED ISSUE #1469: Colab/Binder links don't work on Github
### Describe the bug

Our cloud runner links don't work on GitHub. E.g. https://github.com/stellargraph/stellargraph/blob/develop/demos/calibration/calibration-pubmed-link-prediction.ipynb

Clicking either button hits:

```
Refused to load https://colab.research.google.com/github/stellargraph/stellargraph/blob/master/demos/calibration/calibration-pubmed-link-prediction.ipynb because it does not appear in the frame-src directive of the Content Security Policy.
```

```
Refused to load https://mybinder.org/v2/gh/stellargraph/stellargraph/master?urlpath=lab/tree/demos/calibration/calibration-pubmed-link-prediction.ipynb because it does not appear in the frame-src directive of the Content Security Policy.
```

in Safari's web console.

### Environment

StellarGraph eb4d832c14e8bc8180efb96b134c892d868a890d
### Additional context

N/A

### PR DESCRIPTION
**Pull request by timpitman**
This fixes #1469 - Github links to cloud runners were broken due to https://github.com/stellargraph/stellargraph/pull/1398.

This PR reverts #1398 and adds cloud runner links to sphinx using prolog and epilog of nbsphinx.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -196,7 +196,11 @@
 .. raw:: html
 
     <div>
-      <p><a href="{{ env.docname.rsplit('/', 1).pop() }}.ipynb" class="btn">Download this notebook</a></p>
+      <p><a href="{{ env.docname.rsplit('/', 1).pop() }}.ipynb" class="btn">Download this notebook</a>
+      <br>Run the master version of this notebook:
```
Comment (huonw) [inline docs/conf.py:200]: I wonder if this would be better all on one line, without the `<br>`?
```suggestion
      Run the master version of this notebook:
```

Comment (huonw) [inline docs/conf.py:200]: Also, it's not directly related to this PR, but I realised that "master version" probably doesn't mean much to a user, an alternative would be something like:

```suggestion
      <br>Run the latest release of this notebook:
```

Future work along this line would be changing it to "Run this notebook", by using something like nbsphinx approach. They use `env.config.release` to have each doc version page execute with exactly that git tag https://github.com/spatialaudio/nbsphinx/blob/0.6.1/doc/conf.py#L51 .

File: docs/conf.py [hunk 2]
```diff
@@ -196,7 +196,11 @@
 .. raw:: html
 
     <div>
-      <p><a href="{{ env.docname.rsplit('/', 1).pop() }}.ipynb" class="btn">Download this notebook</a></p>
+      <p><a href="{{ env.docname.rsplit('/', 1).pop() }}.ipynb" class="btn">Download this notebook</a>
```
Comment (huonw) [inline docs/conf.py:199]: An alternative to the other two minor tweaks, would be to rephrase this to be one longer line, something like:

```html
<p>
  Execute this notebook: 
  <a href="{{ env.docname.rsplit('/', 1).pop() }}.ipynb" class="btn">Download locally</a>
  <a href="https://mybinder.org/v2/gh/stellargraph/stellargraph/master?urlpath=lab/tree/{{ env.docname }}.ipynb" alt="Open In Binder"><img src="https://mybinder.org/badge_logo.svg"/></a>
  <a href="https://colab.research.google.com/github/stellargraph/stellargraph/blob/master/{{ env.docname }}.ipynb" alt="Open In Colab"><img src="https://colab.research.google.com/assets/colab-badge.svg"/></a>
</p>
```

That looks something like:

image

File: docs/conf.py [hunk 3]
```diff
@@ -196,7 +196,11 @@
 .. raw:: html
 
     <div>
```
Comment (huonw) [inline docs/conf.py:198]: Oh, one more thing; if we add
```suggestion
    <div class="admonition info">
```

Then these boxes are highlighted slightly more separately:

image

File: docs/conf.py (hunk 2 above)
Comment (timpitman) [reply docs/conf.py:199]: I like this version grouping them

File: docs/conf.py (hunk 1 above)
Comment (timpitman) [reply docs/conf.py]: "latest release" does read better - have adopted in the notebooks themselves; in sphinx I've adopted your suggestion below.

File: docs/conf.py (hunk 3 above)
Comment (timpitman) [reply docs/conf.py]: I added this; can't decide if it's needed with the other change though?

Comment (huonw) [reply docs/conf.py]: I think it helps especially at the end at the end to avoid the text running into the action buttons.
